tests/test6.c

```diff
@@ -1,15 +1,27 @@
+
+#define uthash_ptrdiff_t long
+
+#define uthash_noheaders
 #include "uthash.h"
+#include <string.h>
 #include <stdlib.h>   /* malloc */
 #include <stdio.h>    /* printf */
 
 /* Set up macros for alternative malloc/free functions  */
 #undef uthash_malloc
 #undef uthash_free
+#undef uthash_strlen
+#undef uthash_memcmp
 #define uthash_malloc(sz) alt_malloc(sz)
 #define uthash_free(ptr,sz) alt_free(ptr)
+#define uthash_strlen(s) alt_strlen(s)
+#define uthash_memcmp(v1,v2,s) alt_memcmp(v1,v2,s)
+
+static void * search_key = 0;
+static int seen_strlen = 0;
 
 typedef struct example_user_t {
-    int id;
+    char key[3];
```

Comment:
Aha. Okay, then.

Tangent: I assume that none of the test cases compile on your target platform because they *all* use stdio, is that right? Do you have any interest in trying to get the test cases to build on your platform, and/or ideas about what that would look like?